include/lsst/jointcal/ProperMotion.h

```diff
@@ -0,0 +1,84 @@
+// -*- LSST-C++ -*-
+/*
+ * This file is part of jointcal.
+ *
+ * Developed for the LSST Data Management System.
+ * This product includes software developed by the LSST Project
+ * (https://www.lsst.org).
+ * See the COPYRIGHT file at the top-level directory of this distribution
+ * for details of code ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef LSST_JOINTCAL_PROPER_MOTION_H
+#define LSST_JOINTCAL_PROPER_MOTION_H
+
+#include <iostream>
+#include <memory>
+#include <math.h>  // atan2
+
+namespace lsst {
+namespace jointcal {
+
+class Point;
+
+/**
+ * Proper motion data for a reference star or fitted star.
+ *
+ * Whether to just use these values or fit them is determined by the RefStar and FittedStar they belong to.
+ *
+ * Units are radians/year
+ * Note: RA proper motion is `pm_ra*cos(dec)`
+ */
+class ProperMotion {
+public:
+    ProperMotion(double ra, double dec, double raErr, double decErr, double raDecCov = 0)
+            : _ra(ra), _dec(dec), _raErr(raErr), _decErr(decErr), _raDecCov(raDecCov) {
+        _offsetBearing = atan2(dec, ra);
+    };
+
+    // PM data is stored as unique_ptr: move when assigning from C++, and copy from python.
+    ProperMotion(ProperMotion const &) = default;
+    ProperMotion(ProperMotion &&) = default;
+    ProperMotion &operator=(ProperMotion const &) = delete;
+    ProperMotion &operator=(ProperMotion &&) = delete;
+    ~ProperMotion() = default;
```

Comment:
I can't see the point in deleting some of these. (Do we need to say explicitly `=default` at all?) 

This is a simple "struct" storing 6 double values, isn't it? Having noncopy-able unique_ptr-s pointing to a ProperMotion instance has nothing to do with the copyability of this class.

Reply:
Yes, we need to explicitly declare or delete constructors/destructors: https://developer.lsst.io/cpp/style.html#b-classes-should-explicitly-declare-or-delete-compiler-generated-members

It's not just a struct: `apply` does real work, and it should eventually have code related to the covariance calculation.

We don't actually have a policy on whether `delete` is the right choice if the expected usage doesn't actually need that move/copy operation. I'll ask on #software-dev

Reply:
Following up from discussion on slack (see #software-dev) and video chat: I'm making these all `default` since there's no harm in having them, and although C++ guidance overall has shifted away from explicitly defining all `default` members, our stack guideline hasn't changed yet.